internal/dag/httpproxy_processor.go

```diff
@@ -686,33 +686,36 @@ func (p *HTTPProxyProcessor) processHTTPProxyTCPProxy(validCond *contour_api_v1.
 	return ok
 }
 
-// validHTTPProxies returns a slice of *contour_api_v1.HTTPProxy objects.
-// invalid HTTPProxy objects are excluded from the slice and their status
-// updated accordingly.
-func (p *HTTPProxyProcessor) validHTTPProxies() []*contour_api_v1.HTTPProxy {
-	// ensure that a given fqdn is only referenced in a single HTTPProxy resource
+// listRootProxies returns a slice of root *contour_api_v1.HTTPProxy
+// objects, where "root objects" is defined by the presence of the
+// .Spec.VirtualHost.Fqdn field.
+// Proxies with conflicting FQDNs are excluded from the result, and
+// have their status updated with the corresponding error.
+func (p *HTTPProxyProcessor) listRootProxies() []*contour_api_v1.HTTPProxy {
+	// Ensure that a given FQDN is only referenced in a single HTTPProxy resource.
 	var valid []*contour_api_v1.HTTPProxy
-	fqdnHTTPProxies := make(map[string][]*contour_api_v1.HTTPProxy)
+	roots := make(map[string][]*contour_api_v1.HTTPProxy)
+
 	for _, proxy := range p.source.httpproxies {
 		if proxy.Spec.VirtualHost == nil {
 			valid = append(valid, proxy)
```

Comment:
This adds non-root proxies to the return list too which contradicts the new name/comments.

Reply:
Ah I missed that; good point.